zndraw/server/events.py

```diff
@@ -522,11 +557,82 @@ def scene_update(data: SceneUpdateData):
 
 
 @io.on("selection:run")
-def selection_run(data: SelectionRunData):
+def selection_run(data: dict):
     """Run the selection."""
-    tasks.run_selection.delay(request.url_root, session["token"], data)
+    tasks.run_selection.delay(
+        f"http://127.0.0.1:{current_app.config['PORT']}", session["token"], data
+    )
 
 
 @io.on("scene:trash")
 def scene_trash():
-    tasks.scene_trash.delay(request.url_root, session["token"])
+    tasks.scene_trash.delay(
+        f"http://127.0.0.1:{current_app.config['PORT']}", session["token"]
+    )
+
+
+@io.on("analysis:run")
+def analysis_run(data: dict):
+    """Run the analysis."""
+    io.emit(
+        "analysis:run:enqueue", to=f"webclients_{session['token']}", include_self=False
+    )
+    tasks.run_analysis.delay(
+        f"http://127.0.0.1:{current_app.config['PORT']}", session["token"], data
+    )
+
+
+@io.on("modifier:run")
+def modifier_run(data: dict):
+    """Run the modifier."""
+    io.emit(
+        "modifier:run:enqueue", to=f"webclients_{session['token']}", include_self=False
+    )
+    tasks.run_modifier.apply_async(
+        (f"http://127.0.0.1:{current_app.config['PORT']}", session["token"], data)
+    )
+
+
+@io.on("modifier:register")
+@typecast
+def modifier_register(data: ModifierRegisterData):
+    """Register the modifier."""
+
+    if data.default:
+        MODIFIER_HOSTS = cache.get("MODIFIER_HOSTS")
+        MODIFIER_SCHEMA = cache.get("MODIFIER_SCHEMA")
+    else:
+        MODIFIER_HOSTS = cache.get("ROOM_MODIFIER_HOSTS").get(session["token"], {})
+        MODIFIER_SCHEMA = cache.get("ROOM_MODIFIER_SCHEMA").get(session["token"], {})
```

Comment:
This would be simplified if we removed `ROOM_MODIFIERS_HOSTS`. Also, a minor thing but it feels weird to use capital letters for the `MODIFIER_HOSTS` variables when it can point to different things in this function. It indicates a constant which it is not here.

Reply:
I wanted a way to highlight the information taken from the cache but I fully agree that all caps is a bad idea.